fashion-hyperparameter/optimize_class.py

```diff
@@ -0,0 +1,106 @@
+import os
+from collections import defaultdict
+from itertools import tee
+import json
+
+from jina.flow import Flow
+from jina.helper import colored
+from jina.logging import default_logger as logger
+
+
+class Optimizer:
+    def __init__(self, index_yaml, evaluate_yaml, 
+                       index_document_generator, evaluation_document_generator,
+                       index_batch_size, query_batch_size,
+                       trial_parameter_sampler,
+                       n_trials,
+                       direction='maximize', seed=42,
+                       config_dir='config', best_config_filename='best_config.json'):
+        self.index_yaml = index_yaml
+        self.evaluate_yaml = evaluate_yaml
+        self.index_document_generator = index_document_generator
+        self.evaluation_document_generator = evaluation_document_generator
+        self.index_batch_size = index_batch_size
+        self.query_batch_size = query_batch_size
+        self.trial_parameter_sampler = trial_parameter_sampler
+        self.n_trials = n_trials
+        self.direction = direction
+        self.seed = seed
+        self.config_dir = config_dir
+        self.best_config_filename = best_config_filename
+
+
+    def run_indexing(self):
```

Comment:
Just a note for future design, evaluation is designed to be `flow` agnostic (in theory one can evaluate any flow (even an Index one)). So the final version should not assume that we are running `indexing` and later `querying`. I think this `indexing` part should be like a `pre_hook` callback or something.

Reply:
This idea might be too advanced for our users. Not sure about it. We can rename the `run_querying` to `run_evaluation`